Chapter 4 Exercises: 

Algorithm Workbench #3

Write a FOR loop that displays the following sets of numbers: 0,10,20,30,40,50 ... 1000

In [6]:
# First we want to write the for clause for the range and how we display the numbers through a 1000 by 10s
for i in range(0, 1001, 10):
# Next we will write the if-else statement to display what a number would input past our range and within our range:   
    if i != 1000:
        print(i, end=', ')
    else:
        print(i)


0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260, 270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370, 380, 390, 400, 410, 420, 430, 440, 450, 460, 470, 480, 490, 500, 510, 520, 530, 540, 550, 560, 570, 580, 590, 600, 610, 620, 630, 640, 650, 660, 670, 680, 690, 700, 710, 720, 730, 740, 750, 760, 770, 780, 790, 800, 810, 820, 830, 840, 850, 860, 870, 880, 890, 900, 910, 920, 930, 940, 950, 960, 970, 980, 990, 1000


Algorithm Workbench #4

Write a loop that asks the user to enter a number. The loop should iterate 10 times and keep a running total of the numbers entered. 

In [12]:
# First we want to name the variable total for our running total of the numbers entered:
Total = 0
# Next we want to specify that we want to run our loop 10 times:
for i in range(10):
# Now we can ask the user to enter a number using the input function  
    Number = float(input("Enter a number: "))
# Our final step is to write a statement that assigns the value to the total+number to the total, using an augmented assignment operator:    
    Total = Total+Number
# Then we can ask to display the output of our code: 
print("The total sum of the numbers entered:", Total)


Enter a number:  2.2
Enter a number:  3
Enter a number:  4.2
Enter a number:  5.5
Enter a number:  6
Enter a number:  2.4
Enter a number:  1
Enter a number:  7
Enter a number:  9
Enter a number:  5


The total sum of the numbers entered: 45.3


Programming Exercises #1

A bug collector collects bugs every day for five days.
Write a program that keeps a running total of the number of bugs collected during the five days.
The loop should ask for the number of bugs collected for each day


In [18]:
# First we want to name the variable total bugs for our running total of the number of bugs collected:
Total_bugs = 0
# Next we want to specify that we want to run our loop for every 5 days:
for day in range(1, 6):
# Now we can ask the user the number of bugs collected on the specific days using the input function:   
    Bugs_collected = int(input("Enter the number of bugs collected on day {}: ".format(day)))
# Then we write the statement that assigns the value to the total+number to the total of number bugs collected:   
    Total_bugs = Total_bugs+Bugs_collected
# Lastly we can ask to display the output of number bugs collected in the five days:
print("Total number of bugs collected in five days:", Total_bugs)


Enter the number of bugs collected on day 1:  23
Enter the number of bugs collected on day 2:  13
Enter the number of bugs collected on day 3:  11
Enter the number of bugs collected on day 4:  18
Enter the number of bugs collected on day 5:  21


Total number of bugs collected in five days: 86


Programming Exercise #4

The distance a vehicle travels can be calculated as follows: distance=speedxtime 
For example, if a train travels 40 miles per hour for three hours, the distance traveled is 120 miles.
Write a program that asks the user for the speed of a vehicle (in miles per hour) and the number of hours it has traveled.

In [25]:
# First we want to ask the user the speed of the vehicle in MPH and the hours in which the vehicle has traveled using the input function: 
# We will want to use float for the speed because of the decimal but we can keep it simple for the hours and keep it to whole hours:
speed = float(input("Enter the speed of the vehicle (in MPH): "))
hours = int(input("Enter the number of hours the vehicle has traveled: "))
# Next we can ask to print the hour and distance traveled:
print("Hour\Distance Traveled")
print("-------------------------")
# Next we can use the for clause to get the range we want:
for hour in range(1, hours + 1):
# Now we can write the formula and use the hour for the time:   
    distance = speed * hour
# We can ask it to display the hour and distance that the vehicle has traveled:    
    print(hour, distance)


Enter the speed of the vehicle (in MPH):  45
Enter the number of hours the vehicle has traveled:  4


Hour\Distance Traveled
-------------------------
1 45.0
2 90.0
3 135.0
4 180.0


Programming Exercise #12


Write a program that predicts the approximate size of a population of organisms. 
The application should prompt the user to enter the starting number of organisms, the average daily population increase (as a percentage), and the number of days the organisms will be left to multiply.

In [33]:
# First we want to ask the user to enter the starting number of organisms using the input function:
Start_Population = int(input("Enter the starting number of organisms: "))
# Next we can ask the user the average daily population increase as a percentage using the input function:
Daily_Population_Increase = float(input("Enter the average daily population increase (as a percentage): "))
# Now we can also include the number of days the organism will be left to multiply using the input function:
Days = int(input("Enter the number of days the organisms will be left to multiply: "))
# Now we can assign the population to the starting number the user inputed and continue with our program:
Approximate_Population = Start_Population
# Next we can ask to print the day with the approximate population
print("\nDay\tApproximate_Population")
print("----------------")
# Lastly we can use the for clause to range the number of days:
for day in range(1, Days + 1):
# Then we will ask it to display our results in the table:   
    print(day, "\t", Approximate_Population)   
    Approximate_Population *= (1 + Daily_Population_Increase / 100)


Enter the starting number of organisms:  44
Enter the average daily population increase (as a percentage):  3
Enter the number of days the organisms will be left to multiply:  6



Day	Approximate_Population
----------------
1 	 44
2 	 45.32
3 	 46.6796
4 	 48.079988
5 	 49.52238764
6 	 51.0080592692


Programming Exercise #15

In this chapter, you saw an example of a loop that draws a square. Write a turtle graphics program that uses nested loops to draw 100 squares

In [43]:
# Had to ask AI for help with this one. I was mainly confused on how you asked how big you wanted your square to be and the set up besides the fact that it was 100 squares:
import turtle

# Create a new turtle screen and set its background color with plain white:
screen = turtle.Screen()
screen.bgcolor("white")

# Create a new turtle named "square"
square = turtle.Turtle()

# Set the speed of the turtle
square.speed(10)

# Next we will use a for clause or a for loop to insure we have 100 squares:
for i in range(100):
# Use another for loop to draw the next square:
  for j in range(4):
      square.forward(100) 
  square.right(90)     
# Now we will move the turtle to a new position without drawing anything
  square.penup()
  square.forward(10)  
  square.pendown()

# Lastly we hide the turtle/close the turtle graphics window
square.hideturtle()
turtle.done()

TclError: no display name and no $DISPLAY environment variable